src/lib/SvelteVirtualList.svelte

```diff
@@ -514,6 +578,49 @@
         }
     })
 
+    // Fill the viewport with items if infinite scrolling is enabled and there are fewer items than the viewport size.
+    $effect(() => {
+        if (infiniteScrollEnabled && infiniteScrollCallback && initialized && viewportElement && !isListComplete && !isLoadingData) {
+            const scrollHeight = viewportElement.scrollHeight
+            const clientHeight = viewportElement.clientHeight
+            const currentTime = Date.now()
+
+            // Reset call count if enough time has passed (reset after infiniteLoopPreventionTimeout)
+            if (currentTime - onReachEndLastCallTime > infiniteScrollLoopPreventionTimeout) {
+                onReachEndCallCount = 0
+            }
+
+            // Prevent infinite loops by limiting calls
+            if (onReachEndCallCount >= infiniteScrollLoopPreventionCallCount) {
+                console.warn(
+                    'SvelteVirtualList: onReachEnd has been called too many times in a short period. ' +
+                        'This usually indicates an infinite loop. Disabling further calls. ' +
+                        'Check your onReachEnd implementation to ensure it properly loads data or returns true when complete.'
+                )
+                isListComplete = true
+                return
+            }
+
+            // Check if content doesn't fill viewport
+            if (scrollHeight <= clientHeight + infiniteScrollThreshold) {
+                onReachEndCallCount++
+                onReachEndLastCallTime = currentTime
+
+                isLoadingData = true
+                const result = infiniteScrollCallback()
+                if (result instanceof Promise) {
+                    result.then((complete: boolean) => {
+                        isListComplete = complete
+                        isLoadingData = false
+                    })
+                } else {
+                    isListComplete = result
+                    isLoadingData = false
+                }
+            }
+        }
+    })
+
```

Comment:
This code is supposed to loop through and call the scroll method until the list is complete, or until we've got more items than the viewport can support (so the items don't stop partway through the list on a really big browser tab). I've had a lot of trouble with this however and I haven't had time to debug it yet.